core/src/main/java/com/sequenceiq/cloudbreak/repository/cluster/ClusterTemplateViewRepository.java

```diff
@@ -21,8 +21,18 @@
             "WHERE b.workspace.id= :workspaceId AND b.status <> 'DEFAULT_DELETED'")
     Set<ClusterTemplateView> findAllActive(@Param("workspaceId") Long workspaceId);
 
-    @Query("SELECT c FROM ClusterTemplateView c WHERE c.stackTemplate.environmentCrn= :environmentCrn AND c.status <> 'DEFAULT_DELETED'")
-    Set<ClusterTemplateView> getAllByEnvironmentCrn(@Param("environmentCrn") String environmentCrn);
+    // Need to fetch template which are defaults and which are env related. If the Env has no DL added then we will
+    // show all the template in the actual provider context
+    @Query("SELECT c FROM ClusterTemplateView c "
+            + "LEFT JOIN FETCH c.stackTemplate s "
+            + "WHERE (s.environmentCrn= :environmentCrn OR c.status = 'DEFAULT') "
+            + "AND c.status <> 'DEFAULT_DELETED' "
+            + "AND ((c.cloudPlatform = :cloudPlatform AND :cloudPlatform IS NOT NULL) OR (:cloudPlatform IS NULL)) "
+            + "AND ((c.clouderaRuntimeVersion = :runtime AND :runtime IS NOT NULL) OR (:runtime IS NULL))")
+    Set<ClusterTemplateView> getAllByEnvironmentCrn(
```

Comment:
The name should describe that this will return only with non deleted defaults and user managed ones by optional cloud platform and runtime. Not just a simply get all by env crn :)